cloud/docker_client.go

```diff
@@ -119,12 +125,52 @@ func (c *dockerClientImpl) EnsureImageDownloaded(ctx context.Context, h *host.Ho
 		// image already exists
 		return imageName, nil
 	} else if strings.Contains(err.Error(), "No such image") {
+
 		// image does not exist, import from remote tarball
 		source := types.ImageImportSource{SourceName: url}
+		msg := makeDockerLogMessage("ImageImport", h.Id, message.Fields{
+			"source":     source,
+			"image_name": imageName,
+			"image_url":  url,
+		})
 		_, err = dockerClient.ImageImport(ctx, source, imageName, types.ImageImportOptions{})
 		if err != nil {
 			return "", errors.Wrapf(err, "Error importing image from %s", url)
 		}
+		grip.Info(msg)
+
+		timer := time.NewTimer(0)
+		defer timer.Stop()
+
+		// verify that ImageImport has succeeded
+	retryLoop:
+		for i := 1; i < maxImageInspectAttempts; i++ {
```

Comment:
The way this is written I believe you'll have 1 fewer attempts than the max attempts. However I think a better approach is to set a timeout and check until that timeout